apps/webapp/app/routes/api.v1.store.$action.ts

```diff
@@ -0,0 +1,91 @@
+import type { ActionFunctionArgs } from "@remix-run/server-runtime";
+import { json } from "@remix-run/server-runtime";
+import { KeyValueStoreRequestBodySchema } from "@trigger.dev/core";
+import { z } from "zod";
+import { authenticateApiRequest } from "~/services/apiAuth.server";
+import { logger } from "~/services/logger.server";
+import { KeyValueStore } from "~/services/store/keyValueStore.server";
+import { assertExhaustive } from "~/utils";
+
+const ParamsSchema = z.object({
+  action: z.enum(["GET", "SET", "DELETE"]),
+});
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  logger.info("Key-value store action", { url: request.url });
+
+  // Ensure this is a POST request
+  if (request.method.toUpperCase() !== "POST") {
+    return { status: 405, body: "Method Not Allowed" };
+  }
+
+  const parsedParams = ParamsSchema.safeParse(params);
+
+  if (!parsedParams.success) {
+    logger.info("Invalid params", { params });
+
+    return json({ error: "Invalid params" }, { status: 400 });
+  }
+
+  // Next authenticate the request
+  const authenticationResult = await authenticateApiRequest(request);
+
+  if (!authenticationResult) {
+    logger.info("Invalid or missing api key", { url: request.url });
+    return json({ error: "Invalid or Missing API key" }, { status: 401 });
+  }
+
+  const authenticatedEnv = authenticationResult.environment;
+
+  // Now parse the request body
+  const anyBody = await request.json();
+
+  const bodyObject = z.object({}).passthrough().parse(anyBody);
+
+  const body = KeyValueStoreRequestBodySchema.safeParse({
```

Comment:
Let's make sure we limit the size of the value to something like 256KB, so this isn't used as a DB alternative.